nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutTCP.java

```diff
@@ -0,0 +1,270 @@
+     *            - the current process context.
+     *
+     * @param sessionFactory
+     *            - a factory object to obtain a process session.
+     */
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSessionFactory sessionFactory) throws ProcessException {
+        final ProcessSession session = sessionFactory.createSession();
+        final FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            pruneIdleSenders(context.getProperty(IDLE_EXPIRATION).asTimePeriod(TimeUnit.MILLISECONDS).longValue());
+            context.yield();
+            return;
+        }
+
+        ChannelSender sender = acquireSender(context, session, flowFile);
+        if (sender == null) {
+            return;
+        }
+
+        try {
+            String delimiter = getDelimiter(context, flowFile);
+            ByteArrayOutputStream content = readContent(session, flowFile);
+            if (delimiter != null) {
+                content = appendDelimiter(content, delimiter);
+            }
+            sender.send(content.toByteArray());
+            session.transfer(flowFile, REL_SUCCESS);
+            session.commit();
+        } catch (Exception e) {
+            getLogger().error("Exception while handling a process session, transferring {} to failure.", new Object[] { flowFile }, e);
+            onFailure(context, session, flowFile);
+        } finally {
+            // If we are going to use this sender again, then relinquish it back to the pool.
+            if (!isConnectionPerFlowFile(context)) {
+                relinquishSender(sender);
+            } else {
+                sender.close();
+            }
+        }
+    }
+
+    /**
+     * event handler method to perform the required actions when a failure has occurred. The FlowFile is penalized, forwarded to the failure relationship and the context is yielded.
+     *
+     * @param context
+     *            - the current process context.
+     *
+     * @param session
+     *            - the current process session.
+     * @param flowFile
+     *            - the FlowFile that has failed to have been processed.
+     */
+    protected void onFailure(final ProcessContext context, final ProcessSession session, final FlowFile flowFile) {
+        session.transfer(session.penalize(flowFile), REL_FAILURE);
+        session.commit();
+        context.yield();
+    }
+
+    /**
+     * Helper method to read the FlowFile content stream into a ByteArrayOutputStream object.
+     *
+     * @param session
+     *            - the current process session.
+     * @param flowFile
+     *            - the FlowFile to read the content from.
+     *
+     * @return ByteArrayOutputStream object containing the FlowFile content.
+     */
+    protected ByteArrayOutputStream readContent(final ProcessSession session, final FlowFile flowFile) {
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream((int) flowFile.getSize() + 1);
```

Comment:
I would have liked us to also support taking an extremely large file, separating it based on some incoming delimiter, and sending each message at a time... example being someone wants to send 1GB CSV file over a TCP connection. This processor would require using SplitText before hand so that each FlowFile was only 1 line from the CSV. 

Maybe we can do this as a future enhancement since it would require a bit of work and complexity, see PutSplunk.